packages/descendants/src/Descendants.types.ts

```diff
@@ -0,0 +1,12 @@
+import { ComponentProps } from 'react';
+
+/** */
+export interface Descendant<E extends HTMLElement = HTMLElement> {
+  element: E; // Could also be a RefObject
+  id: string;
+  props?: ComponentProps<any>;
```

Comment:
nice

Reply:
If I can get it to behave, I'd like to pass in the Props type here.

Problem with that though is you'd have to know that type ahead of time when you call `useInitDescendants` so the reducer state gets established with the correct type

Reply:
i think typing this as ComponentProps<any> is ok